File: clientapp/src/app/core/constants/apiV1Routes.ts
import { environment } from "@env";

export default class ApiV1Routes {
  static ApiUrl = environment.baseBackendUrl;
  static Root = "api";
  static Version = "v1";
  static Base =
    ApiV1Routes.ApiUrl + "/" + ApiV1Routes.Root + "/" + ApiV1Routes.Version;

  static Identity = class {
    static Base = ApiV1Routes.Base + "/identity";
    static Login = ApiV1Routes.Base + "/identity/login";
    static Register = ApiV1Routes.Base + "/identity/register";
    static ConfirmEmail = ApiV1Routes.Base + "/identity/confirm";
    static Refresh = ApiV1Routes.Base + "/identity/refresh";
    static AllRoutes = Object.values(this).map(value => `${value}`);
  };

  static Timesheets = class {
    static GetAll = ApiV1Routes.Base + "/timesheets";
    static GetAllByUserId = ApiV1Routes.Base + "/identity/{userId}/timesheets";
    static GetById = ApiV1Routes.Base + "/timesheets/{id}";
    static Create = ApiV1Routes.Base + "/timesheets";
  };
}

File: InternLog.Api/Features/V1/ApiV1Routes.cs
namespace InternLog.Api.Features.V1
{
	public static class ApiV1Routes
	{
		public const string Root = "api";
		public const string Version = "v1";
		public const string Base = Root + "/" + Version;

		public static class Timesheets
		{
			public const string GetAll = Base + "/timesheets";
			public const string GetById = Base + "/timesheets/{id}";
			public const string GetAllByUserId = Base + "/identity/{userId}/timesheets";
			public const string Create = Base + "/timesheets";
			public const string FullUpdate = Base + "/timesheets/{id}";
			public const string Delete = Base + "/timesheets/{id}";
			public const string DeleteAllForUser = Base + "/timesheets/{userId}";
		}

		public static class Identity
		{
			public const string Login = Base + "/identity/login";
			public const string Logout = Base + "/identity/logout";
			public const string Register = Base + "/identity/register";
			public const string UserInfo = Base + "/identity/userinfo";
			public const string ConfirmEmail = Base + "/identity/confirm";
			public const string RefreshToken = Base + "/identity/refresh";
		}

		public static class System
		{
			public const string Check = Base + "/system/check";
		}
	}
}